### ISSUE TITLE
Build appears to timeout at "Collecting info for testables..."

### ISSUE THREAD

**Issue by panicsteve** [travis-ci/travis-ci#3986]
The project in question is pguyot/Einstein.

Looking at the log, the build seems to be going fine. Then the following line is printed:

    [Info] Collecting info for testables...

And then, after a while, Travis aborts, saying it has not had any response back.

    No output has been received in the last ... minutes, this potentially indicates a stalled build or something wrong with the build itself.
    The build has been terminated

Note the "..." minutes.  That is verbatim from the log, by the way.

I'm not sure how to fix the problem since it's not clear to me what Travis is actually doing here that might be taking so long.  The project builds (and tests) fine locally on my Mac.

Thanks for any help!

**Comment by cybernet** [comment]
https://www.traviscistatus.com/incidents/trwr7c8ffjg5

We have received reports of Mac builds stalling partway into execution or being cancelled due to no output. We are tuning the base VM image and its virtual hardware configuration in response, and will provide updates as we roll out changes.

**Comment by jake-b** [comment]
I am seeing this issue on my branch of the same repository. (--branch=ObjCBridging https://github.com/pguyot/Einstein.git) The build seems to go as expected, but then stalls at "Collecting info for testables..."

    No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

I am not sure what is causing the issue or how to correct it.  The xctool command works fine on my local machine.  It builds and all tests are passed.

**Comment by danydev** [comment]
Same here: same command with same xctool version on my mac doesn't hang, but it hangs on travis

**Comment by HongliYu** [comment]
I meet the same problem with xcode7.3

**Comment by erikackermann** [comment]
I am having this issue as well, starting yesterday after updating to `osx_image: xcode7.3`

**Comment by angelolloqui** [comment]
Same problem here with `osx_image: xcode7.3`. We need 7.3 because we have Swift 2.2 in place.

**Comment by solarce** [comment]
Our `xcode7.3` image now contains the GA release. https://blog.travis-ci.com/2016-03-23-xcode-image-updates

**Comment by michaelmerrill** [comment]
Any news on this? We're using `osx_image: xcode7.3` and it's still timing out on `[Info] Collecting info for testables...`.

**Comment by solarce** [comment]
@michaelmerrill news with regards to what? We pushed a 7.3 GA image live on Mar 23rd. We just updated the 7.3 GA image with xctool [0.2.9](https://github.com/facebook/xctool/releases/tag/0.2.9) last night, as well.

**Comment by jake-b** [comment]
I believe this issue was closed in error.  I see this error occurring with 7.2 and I just updated to try a build with 7.3 and I'm seeing the same issue-- Travis hangs at "Collecting Info for Testables":

Example build that fails on 7.2: https://travis-ci.org/pguyot/Einstein/builds/122572626
Example build that fails on 7.3: https://travis-ci.org/pguyot/Einstein/builds/124603262

It seems that several people are still having their builds fail at this point.  It may not be a bug, but it is unexplained behavior and I think the issue should stay open until it can be explained so that we can fix our builds.

Thanks

**Comment by solarce** [comment]
@jake-b I see, I had thought the issue was fixed with the 7.3 GA release of Xcode. I don't have any insight into why builds are hanging with if 7.3 doesn't fix it

**Comment by angelolloqui** [comment]
In our case it was fixed with the 7.3GA. However, we had to switch from xctool to the standard xcodebuild to avoid the error. We will try again with the new xctool deployed this week

**Comment by cotsog** [comment]
@jake-b: Indeed, [as Angel mentionned](https://github.com/travis-ci/travis-ci/issues/3986#issuecomment-212765251), a workaround is to rather use `xcodebuild` rather that `xctool`. We suggest piping the output to `xcpretty` to have, well, prettier logs. Note, however, that `xcpretty` can sometimes hide some errors.

@angelolloqui: `xctool` 0.2.9 has been deployed on our `xcode7.3` image. Can you give it another try?

**Comment by appleguy** [comment]
@cotsog I'm hitting this extensively with the AsyncDisplayKit repository.  We've been having a ton of issues with Travis recently.  Bummer, because it is a gorgeous service and one of the most shining examples of OSS generosity in the industry (seriously).  Let me know if there's anything I can do to address it or if you need more information (ping me at asyncdisplaykit@gmail.com since I often don't see GH notifications)

**Comment by mattmassicotte** [comment]
Also experiencing the hang with xcode7.3, but using xcode7.2 worked fine.

**Comment by jake-b** [comment]
@cotsog Thank you.  I switched to xcodebuild now but still having travis hang.  This time it hangs just after the final "touch" step (touch the .xctests file)  Locally, the tests run immediately after this "touch" step, but on travis it hangs for 10 minutes and fails.

I'm wondering if this is the same issue causing the xctool to hang, or if its unrelated.  I'll keep playing with it.  Thanks.

**Comment by cotsog** [comment]
@jake-b: can you send us a link to a build log showing this behaviour at support (at) travis-ci (dot) com? Thanks!

**Comment by williamsjj** [comment]
@solarce FWIW we don't use Travis but we see a hang on "Collecting info for testables..." in 4 out of every 10 builds we do using xctool since Xcode 7.3.

**Comment by jurezove** [comment]
Seeing this issue as well - will try to switch to xcodebuild.

**Comment by jake-b** [comment]
Dominic from TravisCI support helped me diagnose my issue-- the app requests access to the address book, which brings up a dialog box which is never answered.  

If your tests work locally, but not on Travis, you may have granted access on your local machine, so there is no dialog to answer. The VM, however has the dialog on screen which can't be answered.

I bet there is a whole class of issues like this, requests for location access, etc, which would exhibit this behavior as well.

@panicsteve 's original question is resolved, so this maybe able to be closed... unless others are having a different issue that exhibits the same behavior.

**Comment by jurezove** [comment]
My tests do not present any dialogs whatsoever.

`xcodebuild` works perfectly, so my current (**working**) config is:
````
language: objective-c
osx_image: xcode7.3
script: set -o pipefail && xcodebuild test -workspace [Project].xcworkspace -scheme [Scheme] -sdk iphonesimulator9.3 | xcpretty
````

**Comment by adamrothman** [comment]
Just adding another data point: this is definitely a failing of `xctool` – it happens locally on my development Mac as well as on Travis.

**Comment by mikamboo** [comment]
Same issue with `ng test` of __angular-cli__ version 1.0.0-beta.22-1. The command ignore the `singleRun=true` flag in the karma configuration file. So I modified the test command from `ng test` to `./node_modules/.bin/karma start` and it woks fine.

**Comment by franklinyu** [comment]
@adamrothman You may want to describe your experience in facebook/xctool#702, to xctool guys who don't believe xctool is to blame.
